### PR TITLE
Fix #9 Support BigDecimal as javaType

### LINKED ISSUE #9: Support BigDecimal as javaType
I tried `BigDecimal` as `javaType` of `<Attribute>`. A generated Java code seems good, but generated Scala code is using `String` instead of `BigDecimal`.

### PR DESCRIPTION
**Pull request by laysakura**
# The purpose / the description

- support BigDecimal as javaType to fix #9

# Checklist for a pull request

- [x] Have you read [the guideline for contributors](https://github.com/folio-sec/reladomo-scala/blob/master/CONTRIBUTING.md)?
- [x] Do all the commits contain `Sign-off by: Full Name <email address>` line?
  - `git commit --signoff` or `git commit -s` that will add a line

### CODE REVIEW

File: sbt-reladomo-plugin/src/test/resources/src_unmanaged/kata/domain/scala_api/AllTypes.scala [hunk 1]
```diff
@@ -55,10 +55,10 @@ case class NewAllTypes(id: Int, booleanValue: Boolean, byteValue: Byte, shortVal
...
       case _ => underlyingObj.setNullableDoubleValueNull()
     }
-    underlyingObj.setNullableDateValue(nullableDateValue.orNull[java.util.Date])
-    underlyingObj.setNullableTimestampValue(nullableTimestampValue.orNull[java.sql.Timestamp])
-    underlyingObj.setNullableStringValue(nullableStringValue.orNull[String])
-    underlyingObj.setNullableByteArrayValue(nullableByteArrayValue.orNull[Array[Byte]])
+    underlyingObj.setNullableDateValue(nullableDateValue.map(_nullableDateValue => _nullableDateValue).orNull[Date])
+    underlyingObj.setNullableTimestampValue(nullableTimestampValue.map(_nullableTimestampValue => _nullableTimestampValue).orNull[Timestamp])
+    underlyingObj.setNullableStringValue(nullableStringValue.map(_nullableStringValue => _nullableStringValue).orNull[String])
+    underlyingObj.setNullableByteArrayValue(nullableByteArrayValue.map(_nullableByteArrayValue => _nullableByteArrayValue).orNull[byte[]])
```
Comment (seratch) [inline sbt-reladomo-plugin/src/test/resources/src_unmanaged/kata/domain/scala_api/AllTypes.scala]:
```scala
scala> None.orNull[byte[]]
<console>:1: error: identifier expected but ']' found.
None.orNull[byte[]]
                 ^
```

`byte[]`  must be `Array[Byte]` in Scala.

File: sbt-reladomo-plugin/src/test/resources/src_unmanaged/kata/domain/scala_api/AllTypes.scala [hunk 2]
```diff
@@ -55,10 +55,10 @@ case class NewAllTypes(id: Int, booleanValue: Boolean, byteValue: Byte, shortVal
       case Some(_nullableDoubleValue) => underlyingObj.setNullableDoubleValue(_nullableDoubleValue)
       case _ => underlyingObj.setNullableDoubleValueNull()
     }
-    underlyingObj.setNullableDateValue(nullableDateValue.orNull[java.util.Date])
-    underlyingObj.setNullableTimestampValue(nullableTimestampValue.orNull[java.sql.Timestamp])
-    underlyingObj.setNullableStringValue(nullableStringValue.orNull[String])
-    underlyingObj.setNullableByteArrayValue(nullableByteArrayValue.orNull[Array[Byte]])
+    underlyingObj.setNullableDateValue(nullableDateValue.map(_nullableDateValue => _nullableDateValue).orNull[Date])
```
Comment (seratch) [inline sbt-reladomo-plugin/src/test/resources/src_unmanaged/kata/domain/scala_api/AllTypes.scala]: In this case, `map(_nullableDateValue => _nullableDateValue)` looks unnecessary.

File: reladomo-scala-common/src/main/scala/com/folio_sec/reladomo/scala_api/package.scala [hunk 3]
```diff
@@ -26,4 +26,6 @@ package object scala_api extends OperationImplicits with AggregateDataImplicits
 
   type Timestamp = java.sql.Timestamp
 
+  type BigDecimal = java.math.BigDecimal
```
Comment (seratch) [inline reladomo-scala-common/src/main/scala/com/folio_sec/reladomo/scala_api/package.scala]: I'm not sure if having a type alias for `java.math.BigDecimal` is really useful. As for `Timestamp`, Reladomo (Java) heavily uses `java.sql.Timestamp`, so I thought that having an alias for it is reasonable.

Comment (laysakura) [reply reladomo-scala-common/src/main/scala/com/folio_sec/reladomo/scala_api/package.scala]: `com.folio_sec.reladomo.generator.MithraObject.Attribute.javaType`  has `"BigDecimal"`, which seems to be derived from schema XML like `<Attribute name="nullablePrice" javaType="BigDecimal"  ... />`.

I added this line In order for `com.folio_sec.reladomo.generator.MithraObject.Attribute.javaType = "BigDecimal"` to be regarded as `java.math.BigDecimal`.

I guess it's better if `com.folio_sec.reladomo.generator.MithraObject.Attribute.javaType` can have "java.math.BigDecimal" while XML has `javaType="BigDecimal"` attribute.
Any idea?

File: sbt-reladomo-plugin/src/test/resources/src_unmanaged/kata/domain/scala_api/AllTypes.scala (hunk 2 above)
Comment (laysakura) [reply sbt-reladomo-plugin/src/test/resources/src_unmanaged/kata/domain/scala_api/AllTypes.scala]: Sure...

Generated from here: https://github.com/folio-sec/reladomo-scala/pull/12/files#diff-b7d7f2dc639b2ac1018db156121ee8c7R682

This code is to convert `Option[ScalaType]` into `Option[JavaType]`.
If `ScalaType => JavaType` requires no code, generated code seems redundant like this case.

If we have strong motivation to keep generated code sophisticated, I would consider better code generation.

File: sbt-reladomo-plugin/src/test/resources/src_unmanaged/kata/domain/scala_api/AllTypes.scala (hunk 1 above)
Comment (laysakura) [reply sbt-reladomo-plugin/src/test/resources/src_unmanaged/kata/domain/scala_api/AllTypes.scala]: Mm... let me think 😭

**Comment by laysakura** [comment]
Current HEAD generates correct codes for our repository using `javaType="BigDecimal"` reladomo-scala.

@seratch pls continue to help me with review comments 🙏 

@itohiro73 Should I squash commits before merge?

File: reladomo-scala-common/src/main/scala/com/folio_sec/reladomo/scala_api/package.scala (hunk 3 above)
Comment (seratch) [reply reladomo-scala-common/src/main/scala/com/folio_sec/reladomo/scala_api/package.scala]: `reladomo-scala-common` is a library which is used runtime. You should not have a type alias which is needed only for the code generator.

File: sbt-reladomo-plugin/src/main/scala/com/folio_sec/reladomo/generator/ScalaCodeGenerator.scala [hunk 4]
```diff
@@ -652,18 +672,32 @@ abstract class ScalaCodeGenerator(mithraObjectXmlPath: String,
...
+          s"    ${underlying}.${toSetterName(attribute.getName)}(${toJavaAttributeCode(
+            attribute.getName,
+            attribute.getJavaType
+          )})"
         }
       }
       .mkString("\n")
   }
 
+  private def toJavaAttributeCode(attributeName: String, attributeJavaType: String): String =
```
Comment (seratch) [inline sbt-reladomo-plugin/src/main/scala/com/folio_sec/reladomo/generator/ScalaCodeGenerator.scala]: I will show an alternative way to do this.

**Comment by seratch** [comment]
@laysakura I just send a pull request toward your branch. Please check it out when you have time. https://github.com/laysakura/reladomo-scala/pull/1

File: reladomo-scala-common/src/main/scala/com/folio_sec/reladomo/scala_api/package.scala (hunk 3 above)
Comment (laysakura) [reply reladomo-scala-common/src/main/scala/com/folio_sec/reladomo/scala_api/package.scala]: Thank you.

I confirmed https://github.com/laysakura/reladomo-scala/pull/1/files#diff-b7d7f2dc639b2ac1018db156121ee8c7R514 solves the problem.

File: sbt-reladomo-plugin/src/test/resources/src_unmanaged/kata/domain/scala_api/AllTypes.scala (hunk 2 above)
Comment (laysakura) [reply sbt-reladomo-plugin/src/test/resources/src_unmanaged/kata/domain/scala_api/AllTypes.scala]: Thanks!
https://github.com/laysakura/reladomo-scala/pull/1/files#diff-b7d7f2dc639b2ac1018db156121ee8c7R687

File: sbt-reladomo-plugin/src/test/resources/src_unmanaged/kata/domain/scala_api/AllTypes.scala (hunk 1 above)
Comment (laysakura) [reply sbt-reladomo-plugin/src/test/resources/src_unmanaged/kata/domain/scala_api/AllTypes.scala]: Thanks!
https://github.com/laysakura/reladomo-scala/pull/1/files#diff-b7d7f2dc639b2ac1018db156121ee8c7R519

File: sbt-reladomo-plugin/src/test/resources/reladomo/issue009/ReladomoClassList.xml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+
+<!--
+  Copyright 2017 Goldman Sachs.
```
Comment (laysakura) [inline sbt-reladomo-plugin/src/test/resources/reladomo/issue009/ReladomoClassList.xml]: @itohiro73 What should I write here?

**Comment by seratch** [comment]
@itohiro73 @laysakura The changes in this pull request look good to me. If there is no concern, let's squash and merge this.

File: sbt-reladomo-plugin/src/test/resources/reladomo/issue009/ReladomoClassList.xml (hunk 5 above)
Comment (itohiro73) [reply sbt-reladomo-plugin/src/test/resources/reladomo/issue009/ReladomoClassList.xml]: This is totally a brand new file and you should put "Copyright 2017 FOLIO Co., Ltd." there.

**Comment by laysakura** [comment]
@seratch @itohiro73 thank you for the review comments 🙏 

pushed final commit.

**Pull request merged**